### PR TITLE
#535 Add material UI into manager

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #535 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 1]
```diff
@@ -98,50 +112,33 @@ class NodeListPage extends React.Component {
             <s.TopWrapper>
               <H2>Nodes</H2>
               <s.NewNodeBtn
-                theme={primaryBtn}
+                variant="contained"
+                color="primary"
                 text="New node"
-                data-testid="new-node"
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Although we have disabled all the tests on this page. But the `data-testid` should be passed down as a prop and so we can use them later in the tests

File: ohara-manager/client/src/components/pages/NodePage/styles.js [hunk 2]
```diff
@@ -33,15 +34,15 @@ export const TopWrapper = styled.div`
   align-items: center;
 `;
 
-export const NewNodeBtn = styled(Button)`
-  margin-left: auto;
-`;
+export const NewNodeBtn = styled(MuiButton)({
```
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/src/components/pages/NodePage/styles.js]: Normally, when writing styles with styled-component we do:

export const NewNodeBtn = styled(Button)`
  margin-left: auto;
`;

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 3]
```diff
@@ -98,50 +112,33 @@ class NodeListPage extends React.Component {
...
+              />
             </Box>
           </s.Wrapper>
-          <NodeNewModal
-            isActive={activeModal === NODE_NEW_MODAL}
-            handleClose={() => {
-              this.setState({ activeModal: null });
-            }}
+          <MuiNewModal
+            open={this.state.newOpen}
```
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: `isOpen` is also okay

File: ohara-manager/client/src/App.js [hunk 4]
```diff
@@ -29,6 +29,13 @@ import LoginPage from 'components/pages/LoginPage';
 import LogoutPage from 'components/pages/LogoutPage';
 import NotFoundPage from 'components/pages/NotFoundPage';
 import { getUserKey } from 'utils/authUtils';
+import { create } from 'jss';
+import { StylesProvider, jssPreset } from '@material-ui/styles';
+
```
Comment (konekoya) [inline ohara-manager/client/src/App.js]: Let's add a comment to explain why we're using jss here. You can probably link to Mui's docs too.

**Comment by konekoya** [comment]
retry manager

File: ohara-manager/client/src/components/pages/NodePage/MuiEditModal.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                    placeholder="node-00"
+                    margin="normal"
+                    fullWidth
+                    variant="outlined"
+                    component={InputField}
+                  />
+
+                  <Field
+                    id="port"
+                    name="port"
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/MuiEditModal.js]: Please add a new prop here -> `label="port"`

File: ohara-manager/client/src/components/pages/NodePage/MuiNewModal.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+                    placeholder="node-00"
+                    margin="normal"
+                    fullWidth
+                    variant="outlined"
+                    component={InputField}
+                  />
+
+                  <Field
+                    id="port"
+                    name="port"
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/MuiNewModal.js]: Please add a new prop here -> `label="port"`

File: ohara-manager/client/cypress/integration/brokerPage.test.js [hunk 7]
```diff
@@ -40,7 +40,7 @@ describe('BrokerPage', () => {
     });
   });
 
-  it('creates a new topic', () => {
+  it.skip('creates a new topic', () => {
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/cypress/integration/brokerPage.test.js]: #1070 is in issue , so now skip this test

File: ohara-manager/client/cypress/integration/workerPage.test.js [hunk 8]
```diff
@@ -16,7 +16,7 @@
 
 import { WORKER } from '../../src/constants/urls';
 
-describe('WorkerPage', () => {
+describe.ship('WorkerPage', () => {
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/cypress/integration/workerPage.test.js]: #1070 is in issue , so now skip this test

File: ohara-manager/client/src/components/pages/NodePage/styles.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/styles.js]: > what's the difference between the previous css declaration vs the current one?

is some

File: ohara-manager/client/cypress/integration/brokerPage.test.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/cypress/integration/brokerPage.test.js]: I think you're referring to a wrong issue?

File: ohara-manager/client/src/components/pages/NodePage/styles.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/src/components/pages/NodePage/styles.js]: Hmm, then can we change this back to the previous style? This is just for the sake of consistency :)

File: ohara-manager/client/src/App.js [hunk 9]
```diff
@@ -29,6 +29,14 @@ import LoginPage from 'components/pages/LoginPage';
...
+//if want to add css to material need add jss Url = https://material-ui.com/customization/css-in-js/#jssprovider
```
Comment (konekoya) [inline ohara-manager/client/src/App.js]: Let's rephrase this to _We need to override Mui with our custom styles. See Mui docs for more info:
https://material-ui.com/customization/css-in-js/#other-html-element

File: ohara-manager/client/cypress/integration/brokerPage.test.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (konekoya) [reply ohara-manager/client/cypress/integration/brokerPage.test.js]: Nevermind, These look like already exist in the master branch.

File: ohara-manager/client/src/components/common/Mui/Form/MuiButton.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      variant={variant}
+      onClick={onClick}
+    >
+      {text}
+    </MuiB>
+  );
+};
+
+MuiButton.propTypes = {
+  text: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Mui/Form/MuiButton.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/styles.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/styles.js]: Done

File: ohara-manager/client/src/components/common/Mui/Form/MuiButton.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import MuiB from '@material-ui/core/Button';
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Mui/Form/MuiButton.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 12]
```diff
@@ -58,9 +59,22 @@ class NodeListPage extends React.Component {
     this.setState({
       activeModal: NODE_EDIT_MODAL,
       activeNode: node,
+      editOpen: true,
     });
   };
 
+  handleNewClickOpen = () => {
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Done

File: ohara-manager/client/src/App.js (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/src/App.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/MuiNewModal.js (hunk 6 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/MuiNewModal.js]: Done

File: ohara-manager/client/src/App.js (hunk 9 above)
Comment (wu87988622) [reply ohara-manager/client/src/App.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/MuiEditModal.js (hunk 5 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/MuiEditModal.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 13]
```diff
@@ -39,6 +38,8 @@ class NodeListPage extends React.Component {
     nodes: [],
     activeModal: null,
     activeNode: null,
+    newOpen: false,
+    editOpen: false,
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Done

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 14]
```diff
@@ -58,9 +59,22 @@ class NodeListPage extends React.Component {
...
 
+  handleNewClickOpen = () => {
+    this.setState({ newOpen: true });
+  };
+
+  handleEditClickOpen = () => {
+    this.setState({ editOpen: true });
+  };
+
+  handleClose = () => {
```
[the remark this answers is not in the document]
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: Done
